---
title: ''
output: pdf_document
---


# Discussion

## Intro
The developed application and the proteomics experiment pipeline demonstrated how interactive engagement with the data not only enhances the experience of biologists but also improves the comprehension of the underlying significance of a biological dataset. By utilizing an entropy-based visualization approach, the conservation of information can be validated, allowing users to select appropriate methods and adjust thresholds, cutoffs, and techniques accordingly. With the selection of factors to indicate over multiple plots, biases can be identified on various levels. Additionally, the analysis can be examined on an individual protein basis, enabling the identification of suspicious results. The statistics module provides users with a simple tool to assess and visualize the results, offering clarity to dense scatter plots through hover functions. Enrichment mapping links statistically significant proteins to either the KEGG pathway database or individual research datasets, depending on user selection. These final results place the data in a biological context, which can be explored interactively in graphical visualizations such as network.

## Mutual Information
Information theory and the concept of mutual information provide a framework for understanding the conservation of entropy in a dataset. Traditionally, the validation of data transformation and normalization methods has been limited to observing distributions through histograms or Quantile-Quantile plots. However, the developed approach goes beyond these conventional methods, providing a deeper context for the variation in the data. By allowing users to choose the method that preserves the maximum amount of information, the approach enables researchers to extract the most valuable insights from their experiments.

## Interactive pipeline
Linear programming scripts are widely utilized for data analysis, and R-shiny provides an efficient platform for implementing these scripts in applications. This approach not only benefits scientists who are not proficient in programming, but it is also convenient for programmers. The developed user interface offers a visually appealing design, and the interactive menus enhance the user experience by eliminating the need to search for specific lines of code on a black screen. Modifying parameters of functions becomes effortless, and data can be visualized without rerunning the entire script. R-shiny executes functions on demand, which means that graphics, variables, and data frames are not recalculated if their dependencies remain unchanged. This feature makes the application both cost-effective and time-efficient.
The analysis pathway is visualized through multiple components, providing users with a comprehensive view of the data. The overview plot, generated using the Q-feature object-oriented programming in the scp R library \citep{Vanderaa2021}, serves as a visual representation of the analysis process. This plot allows users to track the progression of the analysis and identify any missing files or samples. 

## Reporter ion intensity
MS2 reporter ion intensities can be visualized as medians in form of a boxplot, allowing users to assess the distribution of intensities across different factors such as sample type, batch, or channel. This visualization helps identify potential biases or variations in the dataset.

## median coefficient of variation
The reference values of the SCoPE2 publication \citep{Specht2021} could be tested and confirmed. 5 as the number of observation for leading razor peptides was suitable to identify the cutoff of 0.65 by observing whiskers of the boxplots in order to reduce noisy peptides in the dataset.

## Feature wise plot
The feature-wise output further enhances the analysis pathway by enabling users to select individual proteins and explore their associated peptide spectrum, peptide and protein values over the course of the analysis. This feature provides a detailed view of the analysis results, allowing users to detect any missing or erroneous data points, particularly in channels with low or high abundance. By examining these individual protein profiles, users can evaluate the quality of the wet lab work and identify any possible errors or inconsistencies.

## Statistics module
The statistics module within the application provides users with a user-friendly interface for hypothesis testing, even without prior expertise in statistics. By leveraging the powerful R-package limma \citep{Ritchie2015}, the module offers robust statistical analysis methods, empowering users to perform tests and obtain results to validate their hypotheses.One key feature of the statistics module is its ability to handle multiple factors and therefore correct for batch effects. Users can extend the statistical analysis across various factors, allowing them to quantify the impact of different variables and identify differentially expressed proteins. To enhance the user experience, the visualization of statistical results has been improved, resulting in visually appealing and interactive displays. These visualizations enable users to explore and interpret the results effortlessly, aiding in the comprehension of the statistical findings. By providing intuitive and engaging visual representations, the module facilitates the understanding and communication of complex statistical analyses.

All-Round Manipulation of the Actin Cytoskeleton by HIV \citep{Turville2018} example for the most differentially expressed gene. --> elaborate biological meaning.
P68133 (Actin, alpha skeletal muscle)

SAMHD1-Deficient CD14+ Cells from Individuals with Aicardi-GoutiÃ¨res Syndrome Are Highly Susceptible to HIV-1 Infection \citep{Berger2011} example for the most differentially expressed gene. --> elaborate biological meaning.
P31153 (S-adenosylmethionine synthase isoform type-2)


## Enrichment 
The enrichment analysis module within the application addresses the challenge of selecting protein sets for mapping by incorporating a database-fetching functionality. This feature streamlines the enrichment analysis process and eliminates the need for users to manually select sets to map proteins against. The first variant of the enrichment analysis module utilizes the KEGG database, which offers a comprehensive collection of pathways and their interactions. By accessing this database, users can gain valuable insights into the underlying pathways and their relationships. This automated mapping against the KEGG database provides a convenient and efficient way to perform enrichment analysis. The second variant of the module caters to sophisticated users who may have specific protein sets in mind based on their research question. This flexibility allows users to select and incorporate their own protein sets into the analysis, enabling research in multiple contexts and providing a comprehensive insight into the dataset. Both variants of the enrichment analysis module present the results in interactive plots, offering users a dynamic and immersive experience with the data. Users can explore the enriched pathways and interact with the visualizations, gaining a deeper understanding of the relationships and connections within the dataset. Additionally, if desired, users can generate static plots and download them for use in presentations or other offline contexts.

## Limitations 
658 nrows number of proteins in dataset
some samples (=single cells=channels) -> mean missing values 53% 
level of detection very bad --> choice of carrier proteome?



## Future perspectives

Clustering for stratification of cells. 


One future research question could utilize artificial intelligence by predicting cellular behaviour, cell types, or even diseases. Running the application on a server or high-performance cluster would be advantageous to address this question and employ simple machine learning tools such as decision trees, random forests, or support vector machines. The application of artificial intelligence on biological data could reduce the hands-on time by identifying unclear results caused by experimental design, unclean preparation, or incorrect computation.
Another aspect, not limited to research topics, would be to focus on enhancing usability. While a development time of 6 months for a pipeline and user interface is sufficient to meet the analytical needs, there is still room for improvement in terms of user-friendliness. It is crucial to make the application more intuitive and robust to prevent crashes or errors caused by user mistakes. Ensuring a user-friendly experience is particularly important for commercial software, as robustness and ease of use are essential factors for success. However, since all software should be open source and developers should have the freedom to satisfy their curiosity, we can overlook this fact.