---
title: " "
output: pdf_document
header-includes:
  - \usepackage{float}
  - \usepackage{sectsty}
---
# Introduction

## From the Central Dogma of Molecular Biology to Omics and Systems Biology: Exploring Interactions and Quantifying Biological Relevance
Before understanding the underlying principle biological systems undergo and the initial exploration of molecular biology, the processes of live were roughly inferred and not well understood. The central dogma of molecular biology serves as the cornerstone of biological processes, providing a framework for understanding how genetic information is converted into functional proteins. This framework consists of two major processes, which involve transcription, converting DNA into RNA, and translation, converting RNA into functional proteins. In between those processes regulation takes base on a molecular level, by altering structure with chemical adaptations. The regulation is undertaken to provide biological organisms with an highly efficient output of their  genetic products called proteins. Crick's formulation of the central dogma has guided research in molecular biology for over 60 years \citep{Cobb, 2017}. Over the past six decades, researchers have explored various aspects of this principle, collectively referred to as omics, to comprehensively analyze biomolecules in diverse contexts. Omics research has emerged as an approach for comprehensively studying biomolecules and their interactions within biological systems. Omics encompasses various fields, including genomics, transcriptomics, proteomics, metabolomics, and others, each focusing on a specific class of biomolecules. By analyzing multiple omics layers, researchers can gain a holistic view of biological processes, uncover regulatory mechanisms, and identify key players in complex biological networks with computational methods. Systems biology takes a quantitative approach to investigate the interactions, dynamics, and emergent properties of biomolecular networks. Imagining biological systems as complex, multi-layer networks, the ensemble of proteins, known as the proteome, plays a critical role in this interactive structure with many functions such as structural integrity, catalyzing chemical reaction and regulation of cellular funcitons.

## Proteomics
The central dogma of molecular biology offers insights not only at the level of individual terms but also provides a hierarchical understanding of biology. Hierarchical organization is the underlying principle in all topics of biology. Proteins exhibit hierarchical structures that contribute to their functionality. At the primary structure level, proteins are composed of smaller building blocks called amino acids. Combining them results in the protein secondary structure such as alpha helix and beta sheet. These secondary structures further assemble to create higher-order tertiary structures, which represent specific domains within the protein. Finally, the quaternary structure describes the functional state of the protein as a whole at a given time. By studying the proteome, which encompasses all the proteins present in a biological system at a specific moment, we can gain a comprehensive understanding of the hierarchical organization and functional dynamics of biological processes.

### Bulk proteomics
Higher organisms consist of highly differentiated cells in an arrangement called tissue. The term tissue includes skin, muscle or blood and is represented by an ensemble of living cells. Taking a sample of any particular tissue will include all types of cells.








In tissues cells have specific functions and not all proteins are equally expressed in all cells. Taking samples can lead to an effect called averaging over the entire cellular ensemble. Cell sorting techniques like fluorescence activated cell sorting, magnetic bead isolation and liquid chromatography can compensate for this effect and help for studying single cell lines or types.



### Single cell type

### Single cell proteomics




Single-cell proteomics (=SCP) provides a snapshot of the underlying processes in a cell. These proteins can be influenced by various environmental factors and the understanding of abundance or differential expression is crucial to understand the principles of cellular behavior. SCP gives insight in a qualitative but also a quantitative way and can elucidate responses of drugs, infectious diseases, and the development of an organism. In the last 10 years SCP became an indispensable tool for global research communities (Minakshi et al., 2019).



An early approach of qualitative analysis of the cellular proteome involved labeling with fluorescent antibodies and imaging. The major disadvantage of this technique was the limitation to only stain a few proteins per cell. For quantification procedures such as single-cell Western blots, immunoassays or CyTOF have been used. Other disadvantages are the ability to permeate cells, accessibility and binding of the epitope and the creation of specific antibodies for a given protein \citep{Budnik2018}. 

One of those techniques involved RNA-sequencing. Since RNA involves also non-coding RNA, the amount of RNA is often not proportional to the content of proteins in a cell. So the proteinaceous content of a cell was only predicted and quantitative analysis was not possible. 

## Effectors of the proteome

### External effectors

When pathogens interact with cells of an organism a cascade of biochemical reactions takes place. These reactions influence the outcome of the cellular proteome in regards of localization, abundance, post-translational modifications and more (Jean Beltran et al., 2017). 



















































## Methods for Quantifications

A holistic approach to analyze the expression can be sequencing of RNA (RNAseq) or mass-spectrometry (MS).  RNAseq refers to the task of examining the transcriptome of bulk tissue sample  with deep-sequencing technologies. RNA in a sample includes mRNAs, non-coding RNAs, and small RNAs. One of the main disadvantages is, that the mapping relies on the knowledge of the DNA sequence of the observed organism (reference genome) (Wang et al., 2009). Another disadvantage is, that the amount of protein translated from the determined mRNA can only be estimated (Gygi et al., 1999). However, today these predictions can be done with computational methods such as LASSO and the accuracy of the estimate highly depends on selection and validation of this method (Magnusson et al., 2022).

When observing the protein ensemble of a cellular environment with a top-down approach  like DNAseq or RNAseq, the characteristics (e.g. post-translational modifications or charge) and local concentration of the proteins remain hidden. The cause of this phenomenon is that only a fraction of the DNA is transcribed into mRNA and not all mRNA is further translated into protein. Furthermore multiple proteins can be derived from a single DNA-sequence with alternative splicing and different proteins coming from non related sequences acting like building blocks can form protein complexes and/or Protein-RNA complexes. Cells are highly efficient and do not waste energy by producing obsolete proteins, moreover the degradation of proteins is an important energy recovery mechanism. A bottom-up strategy can elucidate these properties of the protein ensemble at a given time leading to proteomics. Early methods in molecular biology were western blots with the disadvantage of having a-priori knowledge of the analyzed protein and quantitative protein assays like the UV-vis-, Bradford- and/or bicinchoninic-acid-assay measuring the entire protein concentration. When studying impact of an external signal or stressor to a cell the abundance and   properties of a protein or protein-ensemble can enlighten the underlying pathway leading to a biological response. Since some proteins do not even function before post-translational modifications happen, a knowledge about these can be crucial for further studies. 


### Mass Spectrometry

Mass spectrometry enables qualitative and quantitative analysis of the entire repertoire of a biological sample. The availability of gene sequences in databases and the ability to match proteins against those sequences with computational methods makes it possible to identify alterations of a sample on a protein level. These alterations can rely on the sequence level or could be to post-translational modifications (PTMs) such as phosphorylation, methylation or else \citep{Aebersold2003}. 


Mass spectrometers measure the mass to charge ratio (m/z) of a particle from a fragmented larger molecule. This is achieved by a physical procedure done with a device which is made of three major components: the ion source, an analyzer, and a detector. The ion sources charges molecules and accelerate them through a magnetic field. The analyzer separates particles according to their mass to charge ratio and the detector senses charged particles and amplifies their signal (Parker et al., 2010). 
MS provides great sensitivity for traces, hence getting an average concentration for all biomolecules in a sample. To identify protein abundance from a single type of cell, cell sorting techniques needed to be improved and applied to MS (Minakshi et al., 2019). Single cell proteomics by mass spectrometry (SCoPE-MS) uses a isobaric labeling with a tandem mass tag (TMT) to enhance signal intensity of a protein species.  Loading of a single-cell channel (SCC) with the right amount can be balanced to the protein concentration of a single cell, leading to single-cell proteomics (Ye et al., 2022). 


MS provides great sensitivity for traces, hence getting an average concentration for all biomolecules in a sample. To identify protein abundance from a single type of cell, cell sorting techniques needed to be improved and applied to MS (Minakshi et al., 2019). Single cell proteomics by mass spectrometry (SCoPE-MS) uses a isobaric labeling with a tandem mass tag (TMT) to enhance signal intensity of a protein species.  Loading of a single-cell channel (SCC) with the right amount can be balanced to the protein concentration of a single cell, leading to single-cell proteomics (Ye et al., 2022).

