---
title: " "
output:
  bookdown::pdf_document2:
    toc: false
#    number_sections: TRUE
geometry: "left = 2.5cm, right = 2cm, top = 2cm, bottom = 2cm"
fontsize: 11pt
header-includes:
  - \usepackage{float}
  - \let\origfigure\figure
  - \let\endorigfigure\endfigure
  - \renewenvironment{figure}[1][2] {
      \expandafter\origfigure\expandafter[H]
    } {
    \endorigfigure
    }
  - \usepackage{sectsty}
  - \usepackage{paralist}
  - \usepackage{setspace}\spacing{1.5}
  - \usepackage{fancyhdr}
  - \usepackage{lastpage}
  - \usepackage{dcolumn}
  - \usepackage{natbib}\bibliographystyle{agsm}
  - \usepackage[nottoc, numbib]{tocbibind}

bibliography: bibliography.bib
---

```{r setup, include=FALSE}
#designed with help of: 
#https://ourcodingclub.github.io/tutorials/rmarkdown-dissertation/

knitr::opts_chunk$set(echo = TRUE)
options(tinytex.verbose = TRUE)

#install dependencies
if (!requireNamespace("tinytex", quietly=TRUE))
  install.packages("tinytex")

library(tinytex)

#install dependencies
if (!requireNamespace("DiagrammeR", quietly=TRUE))
  install.packages("DiagrammeR")

library(DiagrammeR)
```


\allsectionsfont{\centering}
\subsectionfont{\raggedright}
\subsubsectionfont{\raggedright}

\pagenumbering{gobble}

\begin{centering}

\vspace{3cm}
```{r uni_logo, echo=F, out.width="20%"}
knitr::include_graphics("FHCW_logo.png")
knitr::include_graphics("KI_logo.png")
```

\vspace{1cm}

\Large
\doublespacing
{\bf Design and implementation of analysis pipeline for single cell type proteomics data} 
\\ presenting ProteoScanR \& Proteomics Workbench

\vspace{1 cm}

\normalsize
\singlespacing

\vspace{0.5 cm}

\Large

{\bf Lukas Gamp}
\\ 01630003

\vspace{1.5 cm}

In partial fulfillment of the requirement \\for the degree of MSc \\in Bioinformatics

\vspace{1.5 cm}

\normalsize
July 2023

\doublespacing
\doublespacing
Main Supervisor: Dr. DI(FH) Gerhard Duernberger \\
External Supervisor: Assoc. Prof. Ujjwal Neogi, M.Sc. PhD \\
Second examiner: FH-Prof. Dr. Alexandra Graf \\

\end{centering}

\newpage

\pagenumbering{gobble}

\begin{centering}

{\bf Abstract}

\end{centering}

\spacing{1.5}

With the growing prominence of single-cell techniques across various omics fields, there is a pressing need to develop a standardized pipeline for proteomics data in the realm of systems biology. Unlike DNA sequencing and RNA sequencing, proteomics analysis via mass spectrometry incurs high costs in terms of labor and equipment. Additionally, commercially available software solutions often come with hefty price tags and limited transparency regarding the underlying methods employed. However, MaxQuant \citep{Cox2008} presents a promising alternative, complemented by the flexibility using the R programming language. 
Introducing ProteoScanR, a state-of-the-art proteomics pipeline integrated into the user-friendly Proteomics Workbench interface. 
Guided by SCoPE2 \citep{Specht2021, Petelski2021, Vanderaa2021}, the development process of ProteoScanR was thoroughly tested and validated using both bulk and single-cell mass spectrometry data sets. The pipeline implementation in the Proteomics Workbench leverages the power of an interactive environment built in R Shiny, empowering users to discover valuable insights for their specific data set. 
Within the interactive environment, users have the flexibility to customize cutoffs and thresholds for quality control, as well as employ various approaches for data transformation, normalization, missing value imputation, and batch correction. 
ProteoScanR and the Proteomics Workbench served as a valuable tool in guiding the identification of significantly expressed proteins in cells under study. Subsequently, the pathway enrichment analysis provided additional biological contexts for a comprehensive understanding of their functional implications.
The master thesis project serves as a foundation for future advancements in the field of single-cell proteomics. Moreover, the codebase has been designed with robustness and scalability in mind, ensuring ease of maintenance and future expansion of the application.
https://github.com/Lukas67/ProteoScanR

\pagenumbering{roman}

\newpage

\centering
\raggedright
\newpage
\tableofcontents


\newpage

```{r acknowledgments, child='acknowledgments.Rmd'}
```
\pagenumbering{arabic}

\newpage

```{r intro, child = 'introduction.Rmd'}
```

\newpage

```{r methods, child = 'methods.Rmd'}
```

\newpage

```{r results, child = 'results.Rmd'}
```

\newpage

```{r discussion, child = 'discussion.Rmd'}
```

\newpage

```{r conclusion, child = 'conclusion.Rmd'}
```

\newpage

```{r biblio, child = 'bibliography.Rmd'}
```

\newpage

```{r appendix, child = 'appendix.Rmd'}
```

\newpage
\listoffigures
