% Options for packages loaded elsewhere
\PassOptionsToPackage{unicode}{hyperref}
\PassOptionsToPackage{hyphens}{url}
%
\documentclass[
  11pt,
]{article}
\usepackage{amsmath,amssymb}
\usepackage{lmodern}
\usepackage{iftex}
\ifPDFTeX
  \usepackage[T1]{fontenc}
  \usepackage[utf8]{inputenc}
  \usepackage{textcomp} % provide euro and other symbols
\else % if luatex or xetex
  \usepackage{unicode-math}
  \defaultfontfeatures{Scale=MatchLowercase}
  \defaultfontfeatures[\rmfamily]{Ligatures=TeX,Scale=1}
\fi
% Use upquote if available, for straight quotes in verbatim environments
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
\IfFileExists{microtype.sty}{% use microtype if available
  \usepackage[]{microtype}
  \UseMicrotypeSet[protrusion]{basicmath} % disable protrusion for tt fonts
}{}
\makeatletter
\@ifundefined{KOMAClassName}{% if non-KOMA class
  \IfFileExists{parskip.sty}{%
    \usepackage{parskip}
  }{% else
    \setlength{\parindent}{0pt}
    \setlength{\parskip}{6pt plus 2pt minus 1pt}}
}{% if KOMA class
  \KOMAoptions{parskip=half}}
\makeatother
\usepackage{xcolor}
\usepackage[left = 2.5cm, right = 2cm, top = 2cm, bottom =
2cm]{geometry}
\usepackage{graphicx}
\makeatletter
\def\maxwidth{\ifdim\Gin@nat@width>\linewidth\linewidth\else\Gin@nat@width\fi}
\def\maxheight{\ifdim\Gin@nat@height>\textheight\textheight\else\Gin@nat@height\fi}
\makeatother
% Scale images if necessary, so that they will not overflow the page
% margins by default, and it is still possible to overwrite the defaults
% using explicit options in \includegraphics[width, height, ...]{}
\setkeys{Gin}{width=\maxwidth,height=\maxheight,keepaspectratio}
% Set default figure placement to htbp
\makeatletter
\def\fps@figure{htbp}
\makeatother
\setlength{\emergencystretch}{3em} % prevent overfull lines
\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}
\setcounter{secnumdepth}{5}
\usepackage{float}
\usepackage{sectsty}
\usepackage{paralist}
\usepackage{setspace}\spacing{1.5}
\usepackage{fancyhdr}
\usepackage{lastpage}
\usepackage{dcolumn}
\usepackage{natbib}\bibliographystyle{agsm}
\usepackage[nottoc, numbib]{tocbibind}
\ifLuaTeX
  \usepackage{selnolig}  % disable illegal ligatures
\fi
\IfFileExists{bookmark.sty}{\usepackage{bookmark}}{\usepackage{hyperref}}
\IfFileExists{xurl.sty}{\usepackage{xurl}}{} % add URL line breaks if available
\urlstyle{same} % disable monospaced font for URLs
\hypersetup{
  hidelinks,
  pdfcreator={LaTeX via pandoc}}

\author{}
\date{\vspace{-2.5em}}

\begin{document}

\allsectionsfont{\centering}
\subsectionfont{\raggedright}
\subsubsectionfont{\raggedright}

\pagenumbering{gobble}

\begin{centering}

\vspace{3cm}

\includegraphics[width=0.2\linewidth]{FHCW_logo} 
\includegraphics[width=0.2\linewidth]{KI_logo} 

\vspace{1cm}

\Large
\doublespacing
{\bf Design and implementation of analysis pipeline for single cell type proteomics data} 

\vspace{1 cm}

\normalsize
\singlespacing
By

\vspace{0.5 cm}

\Large

{\bf Lukas Gamp}

\vspace{1.5 cm}

in partial fulfilment of the requirement \\for the degree of MSc \\in Bioinformatics

\vspace{1.5 cm}

\normalsize
mm yy
\end{centering}

\newpage

\pagenumbering{gobble}

\begin{centering}

{\bf Abstract}

\end{centering}

\spacing{1.5}

(the spacing is set to 1.5)

no more than 250 words for the abstract

\begin{itemize}
\tightlist
\item
  a description of the research question/knowledge gap -- what we know
  and what we don't know
\item
  how your research has attempted to fill this gap
\item
  a brief description of the methods
\item
  brief results
\item
  key conclusions that put the research into a larger context
\end{itemize}

\pagenumbering{roman}

\newpage

\centering
\raggedright
\newpage
\tableofcontents

\newpage

\section*{Acknowledgements}

Thank you for following this tutorial!

I hope you'll find it useful to write a very professional dissertation.

\newpage

\hypertarget{introduction}{%
\section{Introduction}\label{introduction}}

\hypertarget{proteomics}{%
\subsection{Proteomics}\label{proteomics}}

The proteome is referred to the sum of all proteins of a given sample at
a given time. In the past several quantitative and qualitative assays
were used to enlighten the protein composition of a sample.

An early approach of qualitative analysis of the cellular proteome
involved labeling with fluorescent antibodies and imaging. The major
disadvantage of this technique was the limitation to only stain a few
proteins per cell. For quantification procedures such as single-cell
Western blots, immunoassays or CyTOF have been used. Other disadvantages
are the ability to permeate cells, accessibility and binding of the
epitope and the creation of specific antibodies for a given protein
\citep{Budnik2018}.

One of those techniques involved RNA-sequencing. Since RNA involves also
non-coding RNA, the amount of RNA is often not proportional to the
content of proteins in a cell. So the proteinaceous content of a cell
was only predicted and quantitative analysis was not possible.

\hypertarget{mass-spectrometry}{%
\subsection{Mass Spectrometry}\label{mass-spectrometry}}

Mass spectrometry enables qualitative and quantitative analysis of the
entire repertoire of a biological sample. The availability of gene
sequences in databases and the ability to match proteins against those
sequences with computational methods makes it possible to identify
alterations of a sample on a protein level. These alterations can rely
on the sequence level or could be to post-translational modifications
(PTMs) such as phosphorylation, methylation or else
\citep{Aebersold2003}.

Mass to charge ratio (m/z)

\hypertarget{instrumentation}{%
\subsubsection{Instrumentation}\label{instrumentation}}

\hypertarget{cell-isolation}{%
\paragraph{Cell Isolation}\label{cell-isolation}}

\hypertarget{lysis}{%
\paragraph{Lysis}\label{lysis}}

\hypertarget{digestion}{%
\paragraph{Digestion}\label{digestion}}

\hypertarget{liquid-chromatography}{%
\paragraph{Liquid chromatography}\label{liquid-chromatography}}

In order to separate proteins according to their chemical properties,
size or species a liquid chromatography (LC) is recommended before
ionization.

\hypertarget{ionization}{%
\paragraph{Ionization}\label{ionization}}

In order to analyze a biological sample consisting of proteins in
solution the liquid needs to be vaporized into gas phase. Two techniques
are capable of this procedure. Electrospray ionization (ESI) pushes the
analyte through a capillary and applies an electric current to the
liquid, vaporizing the sample to a charged aerosol. Biomolecules are
fragmented according to their chemical properties and can be further
handled in the mass spectrometer. The fragmented biomolecules are now in
charged droplets separated by their charge on the surface, splitting
further into smaller droplets until they become a gas phase ion. Two
physical models describe the process from gas phase to ion called ``The
ion evaporation model'' (IEM) and ``The charge residue model'' (CRM). In
the ion evaporation model (by Iribarne and Thomoson) the droplets shrink
by evaporation until ions are expelled. The model had its limitation by
explaining same evaporation rate constant among ions with different
chemical properties. In the charge residue model the assumption of one
molecule per droplet leads to an ionization rate constant, which is
independent of the ion itself and relies solely on the generation of the
droplet and the efficiency of the solvent \citep{Wilm2011}.

Matrix-assisted laser desorption/ionization (MALDI)

\hypertarget{coupled-mass-spectrometry-msms}{%
\paragraph{Coupled mass-spectrometry
(MS/MS)}\label{coupled-mass-spectrometry-msms}}

In order to enhance sequence identification, two MS devices are built in
series. In the first run (MS1) the m/z is determined and the molecules
are passed to the next device. Upon passing the molecules are fragmented
into smaller ions and analyzed by the second MS. The fragmentation
highly depends on the chemical bonds found in the molecule. The majority
of these breaks occur on the peptide bond of the protein, although this
is not guaranteed for all bonds and so it can happen that certain
peptide ions have a low abundance \citep{Budnik2018}. These low abundant
peptides will not be detected, hence the problem needs to be faced with
another approach . A solution for this problem is molecular barcoding
with labeling mentioned in the chapter labeling.

\hypertarget{labeling-techniques}{%
\paragraph{Labeling techniques}\label{labeling-techniques}}

For differential analysis proteins need to be labeled to compare mass to
charge intensities in order to quantify observed peptides. Since mass
spectrometry is not a quantitative technique by itself, the peak height
or area does not reflect the abundance of a peptide. Physicochemical
properties of the proteins can change the ionization efficiency and
detectability of the target. However, when comparing the same analyte
between multiple runs of labeled proteins, differences in the mass
spectrum reflect the abundance of those. Labels should be chosen to
change solely the mass of the sample and to not affect folding or other
inherent properties of the protein.

\hypertarget{metabolic-labeling}{%
\paragraph{Metabolic labeling}\label{metabolic-labeling}}

Feeding cells with aminoacids containing heavy isotopes, is the method
of choice in order to label peptides at the earliest possible level.
This atoms can be heavy nitrogen in aminoacids or salts in fertilizer
for plants. Mass shifts are proportional to the isotopes incorporated
during biomass production and are visible after proteolytic cleavage.
Stable isotope labeling in cell culture (SILAC) was presented in the
early 2000s. This method used heavy aminoacid enriched media to feed
cells, in order to quantitatively analyze expression profiles.

\hypertarget{isobaric-labeling}{%
\subparagraph{Isobaric labeling}\label{isobaric-labeling}}

Tandem mass tag (TMT)

Tandem mass tag (TMT) reagents enable to differentiate multiple samples
analyzing in one MS run. The samples are labeled individually and pooled
afterwards, this procedure is called multiplexing. TMTs have the same
charge and differ only by their isotopic masses, the peaks found for
each sample are called reporter ions (RI). Each RI and sample is
interpreted as one channel in downstream analysis. The identification of
these RI leads to an enrichment and identification of low abundance
peptide ions which is common especially in single-cell techniques. With
this technique it is possible to quantify proteins and differ low
abundant proteins from background noise. The disadvantage of isobaric
labeling is, that the co-fragmentation signals can be observed in the
spectrogram and the data needs to be normalized in order to remove
unwanted contribution \citep{Marx2019, Budnik2018}. Furthermore TMTs
have an isotopic distribution according to the distribution found in
nature. This can be corrected during data-acquisition as a defined
spread in other channels.

\hypertarget{ms-spectrum}{%
\subsubsection{MS-Spectrum}\label{ms-spectrum}}

Each peptide is reflected by its` indivual fingerprint in the
ms-spectrum. The fingerprint is based on the chemical properties and
modifications of aminoacids. These aminoacids can be calculated through
their m/z ration and after that interpreted as an aminoacid sequence.
Due to fragmentation of the protein only peptide sequences are visible
in the spectrum. In order to identify proteins, peptides are matched
against a sequence database \citep{Cox2008}. Sequence Databases are
simple .fasta files, which can be downloaded on the uniprot webpage
(www.uniprot.org).

\hypertarget{interpretation-of-the-data}{%
\subsubsection{Interpretation of the
data}\label{interpretation-of-the-data}}

\hypertarget{data-acquisition}{%
\paragraph{Data acquisition}\label{data-acquisition}}

Since ms data has a high resolution, algorithms are used to convert the
raw signal to an interpretable form. MaxQuant is one of many sofware
packages to process the data and provides it for further analysis and
statistical testing. Other software solutions are Protein Discoverer
Thermo Fisher or even packages for R. In this publication we will mainly
focus on the data-acquisition with MaxQuant \citep{Cox2008}.

\hypertarget{three-dimensional-peak-detection}{%
\subparagraph{Three-dimensional peak
detection}\label{three-dimensional-peak-detection}}

The three dimensions of the data are: m/z ratio, intensity and retention
time. The algorithm finds local minima of the function in order to
seperate peaks from each other. The centroid of the peak is detected by
fitting a so called gaussian peak shape fitting. This can be interpreted
as finding the peaks of each m/z spectrum as a function of time.. The
centroid of the peak refers to an isotope.

\hypertarget{deisotoping}{%
\subparagraph{Deisotoping}\label{deisotoping}}

To decrypt the istopic distribution of a biomolecule, MaxQuant creates a
vertex of every single peak and connects them with their possible
isotopic counterparts by finding the proportion of mass of an average
aminoacid to its` respective isotope (averagine \citep{Senko1995}).
Isotoping is the term of such procedure and it is enabled with graph
theory. After this procedure the amount of data points are reduced by a
tenfold and a single peak reflects a small biomolecule.

\hypertarget{label-detection}{%
\subparagraph{Label detection}\label{label-detection}}

The next step in data-acquisition is the detection of labels for
quantification. Isotopic pairs of the label (e.g.~N13, N14, N15)
contained in the tag or aminoacid can be identified by convoluting the
two measured isotope patterns with the theoretical isotope patterns.
With a least-square method the best fit is found iteratively and the
channel/sample can be identified.

\hypertarget{improving-peptide-mass-accuracy}{%
\subparagraph{Improving peptide mass
accuracy}\label{improving-peptide-mass-accuracy}}

The intensity-weighted average of the ms peak centroids (as described in
the 3D peak identification) refers to the mass of the peptide.
Corrections highly depend on the analyzer, MaxQuant uses for an orbitrap
typed analyzer a correction value of 1ppm. Autocorrelation between
centroids is compensated by only using well-identified peptides. As
published the mass precision within a ms experiment ranges around 10e-7.

\hypertarget{peptide-search}{%
\subparagraph{Peptide search}\label{peptide-search}}

Biomolecules can now be searched in a database in forward and reverse
direction. The peptide identification (P-) score indicates the fit of
the data to the found sequence in the database according to the length
of the peptide and is used to calculate the posterior error probability
(FDR). The calculation of the false discovery rate is then calculated by
taking FDR into contrast.

\hypertarget{protein-assembly}{%
\subparagraph{Protein assembly}\label{protein-assembly}}

After these calculation the identified peptides can be aggregated
according to it`s respective protein and quantified. The mentioned
metrics indicating the performance of the peptide search can be used in
downstream analysis. A so called razor peptide indicates the group with
the highest number of identified peptides. Quantification is enabled by
taking only unique peptides into contrast. Posterior error
probabilities, which refers to the chance that the found peptide is a
random event, are multiplied and only distinct sequences with the
highst-scoring are used.

\hypertarget{data-processing}{%
\paragraph{Data processing}\label{data-processing}}

Further analysis can be done with R and respective packages such as
bioconductor. Since there is no state of the art established, analysis
varies upon experimental design.

\hypertarget{reading-the-data}{%
\subparagraph{Reading the data}\label{reading-the-data}}

After processing MaxQuant creates a directory containing all results as
.txt file. The evidence.txt file include all peptide sequence matches
(PSM) with their respective proteins and statistical parameters.

Example fo basic parameters and derivations include:

\begin{itemize}
\tightlist
\item
  Peptide sequence
\item
  Mass to charge ratio (m/z) for all scans (eg. MS1, MS2)

  \begin{itemize}
  \tightlist
  \item
    Mass
  \end{itemize}
\item
  Retention time
\item
  Precursor Ion Fragment

  \begin{itemize}
  \tightlist
  \item
    source of the detected ion also referred as mother ion
  \end{itemize}
\item
  Fraction of total spectrum
\item
  Base peak fraction
\item
  Reporter intensity (RI)

  \begin{itemize}
  \tightlist
  \item
    corrected RI
  \end{itemize}
\item
  Posterior error probability (PEP)
\end{itemize}

\hypertarget{object-oriented-programming}{%
\paragraph{Object oriented
programming}\label{object-oriented-programming}}

In order to streamline the analysis of multiple experiments, object
oriented programming can be applied. The approach in R is to create a so
called Q-feature object, which contains all variables and metadata in a
hierarchical structure. The structure enables sub setting for further
analysis.

\hypertarget{filtering}{%
\paragraph{Filtering}\label{filtering}}

Peptides having a low abundance will be set to zero by the analysis. The
value zero would imply that the sample does not contain the respective
peptide, therefore replacing 0 with not applicable (NA) is crucial for
understanding and interpretation of MS-data.

Peptide sequences matching to the reverse protein sequences (=decoy
database) are considered as possible contaminants. These matches can be
excluded from further analysis.

The author of the SCoPE2 \citep{Petelski2021} pipeline recommends
calculating the sample to carrier ration (SCR). This can be done by
dviding the ion intensity of the sample by the reporter ion intensity of
the carrier channel. However in our experiment no carrier and reference
channel is used.

The next step for quality control is the exclusion of samples with a
high false discovery rate (FDR). When applying multiple statistical
testing (e.g.~t-Test) the obtained p-values can be considered as biased,
because the probability to observe a significant will iteratively
increase with each test performed. Corrections in statistics are an
approach to compensate for the multiplicity of testing. There are many
ways to do this compensation like the Bonferroni method or
Benjamini-Hochberg`s FDR. In Mass spectrometry the common ``way to go''
is calculating a false discovery rate, by dividing false PSMs (=hit of
the decoy database) through the total number of PSMs above the
peptide-spectrum matching score. The peptides spectrum matching score is
defined as -10log10(p). Whereas the p-value is defined that the hit is
done by chance. The calculation of the score is highly dependent on the
data acquisition method used. MaxQuant uses Andromeda , an integrated
search engine. Proteome Discoverer from Thermo Fisher utilizes different
engines such as Mascot or Minora. As published by J.Cox in 2011 Mascot
and Andromeda showed similar performance when comparing FDR values as a
function of coverage. However the observed performance can be lower when
dealing with a decreased coverage \citep{Cox2011}.The threshold for
accepting an FDR of an individual PSM is described as q-value.

Further filter steps can include correction between multiple runs. This
kind of process need the addition of a reference channel. However when
observing a single run, these steps are not crucial for the upcoming
analysis.

\hypertarget{downstream-analysis}{%
\paragraph{Downstream Analysis}\label{downstream-analysis}}

\hypertarget{principle-component-analysis-pca}{%
\subparagraph{Principle component analysis
(PCA)}\label{principle-component-analysis-pca}}

Protein levels can be projected to their principle components (PC) and
clustered to their specific cell type. So cell types are distinguished
by their proteinaceous composition.

\hypertarget{testing-for-differential-expression}{%
\paragraph{Testing for differential
expression}\label{testing-for-differential-expression}}

In order to test for differential expression the package limma from R
bioconductor was used. \citep{Phipson2016}. The package uses a linear
model approach to define a fold expression between sample groups. Before
starting the analysis two matrices need to be computed. The design
matrix identifies samples according to the sample type and defines the
experimental design. In order to create the matrix automatically a
simple algorithm is used within the programmed backend logic. The
expression matrix contains intensities for each identified protein will
be obtained at the end of the pipeline. A function call with the
arguments design and expression matrix calculates the contrast matrix.
The next step is creating a linear model between groups by log-ratios of
their expression values.

\pagenumbering{arabic}

\newpage

\hypertarget{methods}{%
\section{Methods}\label{methods}}

\vspace{0.5cm}

\hypertarget{materials-and-methods}{%
\section{Materials and Methods}\label{materials-and-methods}}

\hypertarget{materials}{%
\subsection{Materials}\label{materials}}

For analysis two types of cells were used. One type is the Jurkat-based
cell line (J-lat) with integrated HIV.

The other type of cells are macrophages with a sample size of 72 cells.
The analysis is done with two groups. A HIV negative (HIV-) control
group and a HIV positive (HIV+) group.

\hypertarget{data}{%
\subsection{Data}\label{data}}

\hypertarget{acquisition}{%
\subsubsection{Acquisition}\label{acquisition}}

Analysis of the data was done with MaxQuant \citep{Cox2008}

\newpage

\hypertarget{results}{%
\section{Results}\label{results}}

Some more guidlines from the School of Geosciences.

This section should summarise the findings of the research referring to
all figures, tables and statistical results (some of which may be placed
in appendices). - include the primary results, ordered logically - it is
often useful to follow the same order as presented in the methods. -
alternatively, you may find that ordering the results from the most
important to the least important works better for your project. - data
should only be presented in the main text once, either in tables or
figures; if presented in figures, data can be tabulated in appendices
and referred to at the appropriate point in the main text.

\textbf{Often, it is recommended that you write the results section
first, so that you can write the methods that are appropriate to
describe the results presented. Then you can write the discussion next,
then the introduction which includes the relevant literature for the
scientific story that you are telling and finally the conclusions and
abstract -- this approach is called writing backwards.}

\newpage

\hypertarget{discussion}{%
\section{Discussion}\label{discussion}}

the purpose of the discussion is to summarise your major findings and
place them in the context of the current state of knowledge in the
literature. When you discuss your own work and that of others, back up
your statements with evidence and citations. - The first part of the
discussion should contain a summary of your major findings (usually 2 --
4 points) and a brief summary of the implications of your findings.
Ideally, it should make reference to whether you found support for your
hypotheses or answered your questions that were placed at the end of the
introduction. - The following paragraphs will then usually describe each
of these findings in greater detail, making reference to previous
studies. - Often the discussion will include one or a few paragraphs
describing the limitations of your study and the potential for future
research. - Subheadings within the discussion can be useful for
orienting the reader to the major themes that are addressed.

\newpage

\hypertarget{conclusion}{%
\section{Conclusion}\label{conclusion}}

The conclusion section should specify the key findings of your study,
explain their wider significance in the context of the research field
and explain how you have filled the knowledge gap that you have
identified in the introduction. This is your chance to present to your
reader the major take-home messages of your dissertation research. It
should be similar in content to the last sentence of your summary
abstract. It should not be a repetition of the first paragraph of the
discussion. They can be distinguished in their connection to broader
issues. The first paragraph of the discussion will tend to focus on the
direct scientific implications of your work (i.e.~basic science,
fundamental knowledge) while the conclusion will tend to focus more on
the implications of the results for society, conservation, etc.

\newpage

\bibliography{bibliography}

\newpage

\hypertarget{appendixces}{%
\section{Appendix(ces)}\label{appendixces}}

\hypertarget{appendix-a-additional-tables}{%
\subsection{Appendix A: additional
tables}\label{appendix-a-additional-tables}}

Insert content for additional tables here.

\newpage

\hypertarget{appendix-b-additional-figures}{%
\subsection{Appendix B: additional
figures}\label{appendix-b-additional-figures}}

Insert content for additional figures here.

\newpage

\hypertarget{appendix-c-code}{%
\subsection{Appendix C: code}\label{appendix-c-code}}

Insert code (if any) used during your dissertation work here.

\end{document}
