---
title: " "
output: 
  pdf_document:
    number_sections: TRUE
geometry: "left = 2.5cm, right = 2cm, top = 2cm, bottom = 2cm"
fontsize: 11pt
header-includes:
  - \usepackage{float}
  - \usepackage{sectsty}
  - \usepackage{paralist}
  - \usepackage{setspace}\spacing{1.5}
  - \usepackage{fancyhdr}
  - \usepackage{lastpage}
  - \usepackage{dcolumn}
  - \usepackage{natbib}\bibliographystyle{agsm}
  - \usepackage[nottoc, numbib]{tocbibind}

bibliography: bibliography.bib
---

```{r setup, include=FALSE}
#designed with help of: 
#https://ourcodingclub.github.io/tutorials/rmarkdown-dissertation/

knitr::opts_chunk$set(echo = TRUE)
options(tinytex.verbose = TRUE)

#install dependencies
if (!requireNamespace("tinytex", quietly=TRUE))
  install.packages("tinytex")

library(tinytex)

#install dependencies
if (!requireNamespace("DiagrammeR", quietly=TRUE))
  install.packages("DiagrammeR")

library(DiagrammeR)
```


\allsectionsfont{\centering}
\subsectionfont{\raggedright}
\subsubsectionfont{\raggedright}

\pagenumbering{gobble}

\begin{centering}

\vspace{3cm}
```{r uni_logo, echo=F, out.width="20%"}
knitr::include_graphics("FHCW_logo.png")
knitr::include_graphics("KI_logo.png")
```

\vspace{1cm}

\Large
\doublespacing
{\bf Design and implementation of analysis pipeline for single cell type proteomics data} 

\vspace{1 cm}

\normalsize
\singlespacing
By

\vspace{0.5 cm}

\Large

{\bf Lukas Gamp}

\vspace{1.5 cm}

in partial fullfilment of the requirement \\for the degree of MSc \\in Bioinformatics

\vspace{1.5 cm}

\normalsize
mm yy
\end{centering}

\newpage

\pagenumbering{gobble}

\begin{centering}

{\bf Abstract}

\end{centering}

\spacing{1.5}

With the growing prominence of single-cell techniques across various omics fields, there is a pressing need to develop a standardized pipeline for proteomics data in the realm of systems biology. Unlike DNA sequencing and RNA sequencing, proteomics analysis via mass spectrometry incurs high costs in terms of labor and equipment. Additionally, commercially available software solutions often come with hefty price tags and limited transparency regarding the underlying methods employed. However, MaxQuant \citep{Cox2008} presents a promising alternative, complemented by the flexibility of post-processing using the R programming language. 
Guided by publications such as SCoPE2 \citep{Specht2021, Petelski2021, Vanderaa2021}, the development process of the proteomics data analysis pipeline was thoroughly tested and validated using both bulk and single-cell mass spectrometry datasets. The pipeline implementation leverages the power of an interactive environment built in R Shiny, empowering users to discover the gold standard for their specific dataset. 
Within the interactive environment, users have the flexibility to customize cutoffs and thresholds for quality control, as well as employ various approaches for data transformation, normalization, missing value imputation, and batch correction. 
The developed application served as a valuable tool in guiding the identification of significantly differentially expressed proteins in HIV cells. Moreover, it facilitated the mapping of these proteins to relevant pathways and provided additional biological contexts for a comprehensive understanding of their functional implications.
The master thesis project serves as a foundation for future advancements in the field of single-cell proteomics. Moreover, the codebase has been designed with robustness and scalability in mind, ensuring ease of maintenance and future expansion of the application.

\pagenumbering{roman}

\newpage

\centering
\raggedright
\newpage
\tableofcontents


\newpage

```{r acknowledgments, child='acknowledgments.Rmd'}
```
\pagenumbering{arabic}

\newpage

```{r intro, child = 'introduction.Rmd'}
```

\newpage

```{r methods, child = 'methods.Rmd'}
```

\newpage

```{r results, child = 'results.Rmd'}
```

\newpage

```{r discussion, child = 'discussion.Rmd'}
```

\newpage

```{r conclusion, child = 'conclusion.Rmd'}
```

\newpage

```{r biblio, child = 'bibliography.Rmd'}
```

\newpage

```{r appendix, child = 'appendix.Rmd'}
```

